File: pdui_flutter/lib/widgets/pdui_text.dart
import 'package:flutter/material.dart';

import '../proto_out/pdui-proto-out.pb.dart';
import 'pdui_widget.dart';

class PduiText extends PduiWidget {
  PBExpression pbText;

  PduiText(this.pbText);

  @override
  Text resolve() {
    return Text(
      pbText.text.data,
      key: pbText.key != "" ? Key(pbText.key) : null,
    );
  }
}

File: pdui-express/lib/widgets/PDUIText.ts
import { PDUIWidget } from "../models/PDUIWidget";
import type { PBExpression, PBText } from "../proto-out/pdui-proto-out.ts";

export class PDUIText extends PDUIWidget {
    data: string;

    constructor({ key, data }: { key?: string; data: string }) {
        super({
            key,
            identity: "text",
            payload: {
                oneofKind: "text",
                text: { data: data },
            },
        });

        this.data = data;
    }
    override toPB(): PBText {
        return {
            data: this.data,
        };
    }

    override toExpression(): PBExpression {
        return {
            key: this.key,
            identity: this.identity,
            payload: {
                oneofKind: "text",
                text: { data: this.data },
            },
        };
    }
}
